### PR TITLE
Add tests for admonitions

### LINKED ISSUE #3496: [DOCUMENTATION] Add new admonitions to telegram classes
### What kind of feature are you missing? Where do you notice a shortcoming of PTB?

I think the docs can do with more explicit types of admonitions. I have a few suggestions:

1. "Returned in":  When viewing a documentation of a class, sometimes I forget which bot method or object through which this class is available.

<details>
<summary> Examples </summary>
<br>

``` python
class ChatInviteLink:
    """
    .. admonition:: Returned In
        :class: returned-in

        :meth:`Bot.create_chat_invite_link`, ...
    """
```

</details>

2. "Available in": Similar to the above, but this would include where and how this object can be found. Motivation: Our docs are currently top-down, i.e. Update -> Message -> Chat, so if I landed in "Update" I can figure out how to get the Chat object, but not vice-versa.  Icon suggestions: [Box-fill](https://icons.getbootstrap.com/icons/box-fill/), [Boxes](https://icons.getbootstrap.com/icons/boxes/), [Geo fill](https://icons.getbootstrap.com/icons/geo-fill/) (liked more by me)

<details>
<summary> Examples </summary>
<br>

``` python
class Chat:
    """
    .. admonition:: Available in
        :class: available-in

        :attr:`telegram.Message.chat`, ...
    """

class Message:
    """
    .. admonition:: Available in
        :class: available-in

        :attr:`telegram.Update.message`, ...
    """
```

</details>

3. "Use in": This would be used in classes which need to be instantiated and then passed into bot methods parameters.
Icon suggestions: [Arrow left](https://icons.getbootstrap.com/icons/arrow-left/), [Paperclip](https://icons.getbootstrap.com/icons/paperclip/), [Funnel](https://icons.getbootstrap.com/icons/funnel/) (liked more by me), [Stack](https://icons.getbootstrap.com/icons/stack/)

<details>
<summary> Examples </summary>
<br>

``` python
class ChatPermissions:
    """
    .. admonition:: Used in
        :class: used-in

        :meth:`Bot.set_chat_permissions`, ...
    """
```

</details>

4. "Shortcuts": This would be used in the bot methods instead of the See Also currently used.

### Describe the solution you'd like

These admonitions are better than "see also" (in most cases) since it tells the user what and why it was linked in the first place.

### Describe alternatives you've considered

see alsos

### Additional context

How to make custom admonitions: https://pradyunsg.me/furo/reference/admonitions/

### PR DESCRIPTION
**Pull request by lemontree210**
adds tests for functionality introduced in #3501 that resolved #3496 

the tests are meant to only run in GH actions before building docs. They are not included in the main test suite, one of the reasons being that Python 3.9+ is required to run the admonition generator.

### CODE REVIEW

File: tests/test_docs_admonition_inserter.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+            assert len(admonition_inserter.admonitions[admonition_type].keys()) > 0
+
+            # keys are telegram classes
+            for cls in admonition_inserter.admonitions[admonition_type]:
+                # Test classes crop up in AppBuilder, they can't come from code being tested.
+                if "tests.conftest" in str(cls):
+                    continue
+
+                assert isinstance(cls, type)
+                assert "telegram" in str(cls), (
```
Comment (Bibo-Joshi) [inline tests/test_docs_admonition_inserter.py:48]: maybe `str(cls).startswith('<class "telegram.")` or something like this? IIRC the format is always the same

File: tests/test_docs_admonition_inserter.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+import telegram.ext
+from docs.auxil.admonition_inserter import AdmonitionInserter
+
+
+@pytest.fixture(scope="session")
+def admonition_inserter():
+    return AdmonitionInserter()
+
+
+class TestAdmonitionInserter:
```
Comment (Bibo-Joshi) [inline tests/test_docs_admonition_inserter.py:31]:
```suggestion
class TestAdmonitionInserter:
    """This is a minimal effort test to ensure that the `AdmonitionInserter` used for automatically inserting references in the docs works as expected. It does not aim to cover all links in the documentation, but rather checks that several special cases are handled correctly which where discovered during the implementation of `AdmonitionInserter`
    """
```

Feel free to rephrase to your liking - a general note in that direction will be useful for future-us though :)

**Review by Bibo-Joshi — COMMENTED** [review]
Left two nitpicks, but the rest LGTM :) Nice follow-up PR!

File: tests/test_docs_admonition_inserter.py (hunk 1 above)
Comment (lemontree210) [reply tests/test_docs_admonition_inserter.py:48]: Done

File: tests/test_docs_admonition_inserter.py [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+
+        assert cls in admonitions[admonition_type]
+
+        # exactly one of the lines in the admonition for this class must consist of the link
+        # (this is a stricter check than just checking if the entire admonition contains the link)
+        lines_with_link = [
+            line
+            for line in admonitions[admonition_type][cls].splitlines()
+            # remove whitespaces and occasional bullet list marker
+            if line.strip().removeprefix("* ") == link
```
Comment (harshil21) [inline tests/test_docs_admonition_inserter.py:156]: `removeprefix` is py3.9+ only, which gives more reason to not have this run with the regular test suite

**Review by harshil21 — COMMENTED** [review]
great. Might I also suggest:

- Moving this test to `tests/docs/` for better organization (and in spirit of #3420)
- Running this test along with GH test documentation build, and excluding this test from the regular test workers.

File: tests/test_docs_admonition_inserter.py (hunk 3 above)
Comment (lemontree210) [reply tests/test_docs_admonition_inserter.py:156]: well yes, the whole admonition generator is 3.9+. Ok, let me figure out how it's done :)

**Comment by lemontree210** [comment]
> great. Might I also suggest:
> 
>     * Moving this test to `tests/docs/` for better organization (and in spirit of #3420)
> 
>     * Running this test along with GH test documentation build, and excluding this test from the regular test workers.

Done. This is the first time I ever changed a GH action, but looks like it works :)

**Review by harshil21 — APPROVED** [review]

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -30,6 +30,8 @@ jobs:
         run: |
           python -W ignore -m pip install --upgrade pip
           python -W ignore -m pip install -r requirements-all.txt
+      - name: Test autogeneration of admonitions
+        run: pytest -v --tb=short tests/docs/admonition_inserter.py
```
Comment (Bibo-Joshi) [inline .github/workflows/docs.yml:34]: what's the difference between using `--tb=short` and not using it?

**Review by Bibo-Joshi — COMMENTED** [review]
Can you maybe add a `README.md` to `tests/docs/` stating that

* that dir contains tests that cover our own scripting logic for the docs
* running `pytest` usually doesn't include that due to the missing `test_` prefix

?

File: .github/workflows/docs.yml (hunk 4 above)
Comment (lemontree210) [reply .github/workflows/docs.yml:34]: The default `auto` mode provides long traceback for the 1st and last entry, and short traceback for other entries ([docs](https://docs.pytest.org/en/6.2.x/usage.html#modifying-python-traceback-printing)). I'm requesting short traceback for all entries. I made sure my exception messages are informative enough.

**Comment by lemontree210** [comment]
> Can you maybe add a `README.md` to `tests/docs/` 

Done

File: .github/workflows/docs.yml (hunk 4 above)
Comment (Bibo-Joshi) [reply .github/workflows/docs.yml:34]: What does 1st and last entry mean here? Do you have an example at hand for comparision?

Comment (lemontree210) [reply .github/workflows/docs.yml:34]: I'm not sure entirely myself, my guess is the entries in a call stack.
[This is the result](https://pastebin.com/5fe6a8Ah) with no `--tb=short` (=`--tb=auto`).
[This is the result](https://pastebin.com/wr4WjSJj) with `--tb=short`.

Comment (Bibo-Joshi) [reply .github/workflows/docs.yml:34]: Okay, that looks sensible to me :D

**Pull request merged**